packages/frontend/src/redux/actions/account.js

```diff
@@ -121,18 +123,23 @@ export const handleRefreshUrl = (prevRouter) => (dispatch, getState) => {
     const currentPage = pathname.split('/')[pathname[1] === '/' ? 2 : 1];
 
     if ([...WALLET_CREATE_NEW_ACCOUNT_FLOW_URLS, WALLET_LOGIN_URL, WALLET_SIGN_URL, WALLET_LINKDROP_URL, WALLET_BATCH_IMPORT_URL, WALLET_VERIFY_OWNER_URL].includes(currentPage)) {
-        const parsedUrl = {
+        let parsedUrl = {
             ...parse(search),
             referrer: document.referrer ? new URL(document.referrer).hostname : undefined,
             redirect_url: prevRouter ? prevRouter.location.pathname : undefined
         };
+        const isPrivateShard = parsedFragment?.calimeroShardId;
+        const parsedFragment = parse(hash);
+        if ( parsedFragment.calimeroShardId) {
+            parsedUrl = { ...parsedUrl, ...parsedFragment };
+        }
```

Comment:
Private shard data is passed as fragment params. This includes transactions if the destination it's a private shard.